src/mito2/src/worker/handle_write.rs

```diff
@@ -202,3 +207,22 @@ fn maybe_fill_missing_columns(request: &mut WriteRequest, metadata: &RegionMetad
 
     Ok(())
 }
+
+/// Updates the next entry id for each region with the returned last entry id contained in the response.
+fn update_next_entry_ids(
+    region_ctxs: &mut HashMap<RegionId, RegionWriteCtx>,
+    response: &AppendBatchResponse,
+) {
+    for (region_id, region_ctx) in region_ctxs.iter_mut() {
+        // Missing a last entry id is not deemed as a critical error and hence no need to abort writing to memtable.
+        let Some(last_entry_id) = response.last_entry_ids.get(&region_id.as_u64()) else {
+            let e = MissingLastEntryIdSnafu {
+                region_id: *region_id,
+            }
+            .build();
+            region_ctx.set_error(Arc::new(e));
+            continue;
```

Comment:
Should we continue or filter out failed regions? cc @evenyag

Reply:
I think it would be a bug when a successful append response is returned with some region's entry id missing.

Reply:
Getting the last entry id must succeed if an OK response is returned. Therefore, we simply unwrap here.

```rust
            match wal_writer.write_to_wal().await.map_err(Arc::new) {
                Ok(response) => {
                    for (region_id, region_ctx) in region_ctxs.iter_mut() {
                        // Safety: the log store implementation ensures that either the `write_to_wal` fails and no
                        // response is returned or the last entry ids for each region do exist.
                        let last_entry_id =
                            response.last_entry_ids.get(&region_id.as_u64()).unwrap();
                        region_ctx.set_next_entry_id(last_entry_id + 1);
                    }
                }
                Err(e) => {
                    // Failed to write wal.
                    for mut region_ctx in region_ctxs.into_values() {
                        region_ctx.set_error(e.clone());
                    }
                    return;
                }
            }
```

Reply:
Add some tests for the underlying log store to ensure this.